plugins/kubernetes/app/models/kubernetes/deploy_yaml.rb

```diff
@@ -46,6 +47,46 @@ def template
       end
     end
 
+    # Sets up the secret_sidecar and the various mounts that are required
+    # if the sidecar service is enabled
+    # /vaultauth is a secrets volume in the cluster
+    # /secretkeys are where the annotations from the config are mounted
+    def set_secret_sidecar
+      pod_volumes =
+        [
+          {name: "secrets-volume", emptyDir: {}},
+          {name: "vaultauth", secret: {secretName: "vaultauth"}},
+          {name: "secretkeys", downwardAPI: {items: [{path: "annotations", fieldRef:{fieldPath:"metadata.annotations"}}]}}
+        ]
+      secret_vol = { mountPath: "/secrets", name: "secrets-volume" }
+      secret_sidecar = {
+        image: ENV.fetch("SECRET_SIDECAR_IMAGE").to_s,
+        name: "secret-sidecar",
+        volumeMounts: [
+          secret_vol,
+          { mountPath: "/vault-auth", name: "vaultauth" },
+          { mountPath: "/secretkeys", name: "secretkeys" }
+        ]
+      }
+      #TODO, do we want resource limits on this, or just give it what it needs?
+      secret_sidecar[:resources] =  { limits: { cpu: 0.1, memory: "100Mi" } }
+
+      # also inject the secrets FS into the primary container so that the
+      # secrets can be shared
+      containers = template.spec.template.spec.containers.dup
```

Comment:
that seems dangerous ... why do we have to dup ? ... prefer a clean container definition ?